componentTemplates/workflows/codeql.yml

```diff
@@ -0,0 +1,58 @@
+name: "CodeQL"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    # The branches below must be a subset of the branches above
+    branches: [ "master" ]
```

Comment:
This needs to be configurable based on what kind of repo it's being added to. I would use `main` as the default with an option to change to `master`.